website/project/tasks.py

```diff
@@ -44,8 +44,7 @@ def update_collecting_metadata(node, saved_fields):
         if node.is_public:
             update_collected_metadata(node._id)
         else:
-            if 'is_public' in saved_fields:
-                update_collected_metadata(node._id, op='delete')
+            update_collected_metadata(node._id, op='delete')
```

Comment:
Now it doesn't look like you're checking for deleted status at all. What if the node is public but it was deleted?

Reply:
I think this means just delete from the ES results. It will just delete it from results [here](https://github.com/CenterForOpenScience/osf.io/blob/develop/website/search/elastic_search.py#L852-L853).